### PR TITLE
Use the TOML file to define dependencies

### PR DESCRIPTION
**Pull request by abrooksv**
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: buildSrc/src/main/kotlin/toolkit-intellij-subplugin.gradle.kts [hunk 1]
```diff
@@ -117,8 +116,9 @@ tasks.runIde {
     }
 }
 
+val versionCatalog = extensions.getByType<VersionCatalogsExtension>().named("libs")
```
Comment (rli) [inline buildSrc/src/main/kotlin/toolkit-intellij-subplugin.gradle.kts:119]:
```suggestion
// TODO: https://github.com/gradle/gradle/issues/15383
val versionCatalog = extensions.getByType<VersionCatalogsExtension>().named("libs")
```

File: buildSrc/src/main/kotlin/toolkit-kotlin-conventions.gradle.kts [hunk 2]
```diff
@@ -14,12 +14,13 @@ plugins {
     id("toolkit-detekt")
 }
 
+// TODO: https://github.com/gradle/gradle/issues/15383
+val versionCatalog = extensions.getByType<VersionCatalogsExtension>().named("libs")
 dependencies {
-    implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlinVersion")
```
Comment (rli) [inline buildSrc/src/main/kotlin/toolkit-kotlin-conventions.gradle.kts:18]: dropped jdk8?

File: buildSrc/src/main/kotlin/toolkit-testing.gradle.kts [hunk 3]
```diff
@@ -14,10 +14,11 @@ plugins {
     id("com.adarshr.test-logger")
 }
 
+// TODO: https://github.com/gradle/gradle/issues/15383
+val versionCatalog = extensions.getByType<VersionCatalogsExtension>().named("libs")
 dependencies {
-    testImplementation("org.mockito.kotlin:mockito-kotlin:$mockitoKotlinVersion")
```
Comment (rli) [inline buildSrc/src/main/kotlin/toolkit-testing.gradle.kts:18]: dropped mockito-kotlin?

**Review by rli — APPROVED** [review]

File: buildSrc/src/main/kotlin/toolkit-kotlin-conventions.gradle.kts (hunk 2 above)
Comment (abrooksv) [reply buildSrc/src/main/kotlin/toolkit-kotlin-conventions.gradle.kts:18]: It is in the Kotlin bundle

```
[bundles]
kotlin = ["kotlin-stdLibJdk8", "kotlin-reflect"]
```

File: buildSrc/src/main/kotlin/toolkit-testing.gradle.kts (hunk 3 above)
Comment (abrooksv) [reply buildSrc/src/main/kotlin/toolkit-testing.gradle.kts:18]: Lives in the bundle Mockito:

```
[bundles]
mockito = ["mockito-core", "mockito-kotlin"]
```

**Pull request merged**